community/kernel/src/main/java/org/neo4j/kernel/impl/store/counts/CountsTracker.java

```diff
@@ -300,56 +449,163 @@ public void close()
         }
     }
 
-    public void rotate( long lastCommittedTxId ) throws IOException
+    @Override
+    protected Object parseHeader( String key, ReadableBuffer value )
```

Comment:
Can we do better for the return type of this method? Why cannot we use generics here?